app/scripts/modules/netflix/ci/services/build.read.service.ts

```diff
@@ -0,0 +1,119 @@
+import {IHttpPromiseCallbackArg, IPromise, module} from 'angular';
+
+import {Api, API_SERVICE} from 'core/api/api.service';
+import {CI_FILTER_MODEL, CiFilterModel} from '../ci.filter.model';
+import {
+  ORCHESTRATED_ITEM_TRANSFORMER,
+  OrchestratedItemTransformer
+} from 'core/orchestratedItem/orchestratedItem.transformer';
+import {SETTINGS} from 'core/config/settings';
+
+interface ICiBuildChangeRef {
+  id: string;
+  displayId: string;
+}
+
+interface ICiBuildChangeTargetAuthor {
+  email: string;
+  name: string;
+  username: string;
+}
+
+interface ICiBuildChangeTarget extends ICiBuildChangeRef {
+  author: ICiBuildChangeTargetAuthor;
+  message: string;
+}
+
+interface ICiBuildChange {
+  from: ICiBuildChangeTarget;
+  ref: ICiBuildChangeRef;
+  to: ICiBuildChangeTarget;
+}
+
+export interface ICiBuild {
+  buildNumber: number;
+  change: ICiBuildChange;
+  completedAt: number;
+  completionStatus: string;
+  endTime?: number;
+  id: string;
+  isRunning?: boolean;
+  repositoryId: string;
+  startedAt: number;
+  startTime?: number;
+}
+
+export interface ICiBuildOutputConfig {
+  data: string[];
+  last: boolean;
+  nextStart: string;
+}
+
+export class CiBuildReader {
+
+  public static get MAX_LINES(): number {
+    return 4095;
+  };
+
+  static get $inject(): string[] {
+    return ['API', 'orchestratedItemTransformer', 'CiFilterModel'];
+  }
+
+  constructor(private API: Api,
+              private orchestratedItemTransformer: OrchestratedItemTransformer,
+              private ciFilterModel: CiFilterModel) {
+  }
+
+  private builds(): any {
+    return this.API.all('ci').all('builds');
+  }
+
+  private transformBuild(build: ICiBuild): ICiBuild {
+    build.startTime = build.startedAt;
+    build.endTime = build.completedAt;
+    build.isRunning = build.completionStatus === 'INCOMPLETE';
+    this.orchestratedItemTransformer.addRunningTime(build);
+
+    return build;
+  }
+
+  public getBuilds(repoType: string, projectKey: string, repoSlug: string): IPromise<ICiBuild[]> {
+    return this.builds()
+      .get({repoType, projectKey, repoSlug, filter: this.ciFilterModel.searchFilter})
+      .then((response: IHttpPromiseCallbackArg<ICiBuild[]>) => {
+        return response.data.map((build: ICiBuild) => this.transformBuild(build));
+      });
+  }
+
+  public getRunningBuilds(repoType: string, projectKey: string, repoSlug: string): IPromise<ICiBuild[]> {
+    return this.builds()
+      .get({repoType, projectKey, repoSlug, completionStatus: 'INCOMPLETE'})
+      .then((response: IHttpPromiseCallbackArg<ICiBuild[]>) => response.data);
```

Comment:
I'll transform the results then.  Since this only populates fields and doesn't manipulate the data returned, this works for me.  Plus, it won't break the existing call.